eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java

```diff
@@ -302,20 +300,11 @@ private SubmittedVariantOperationEntity buildSubmittedOperation(SubmittedVariant
      * variants will be kept unclustered. This potentially will be revisited in the future, but for now (release 2) we
      * are leaving this out of scope.
      */
-    private void checkForMultimaps() {
-        List<String> multimapHashes = new ArrayList<>();
-        for (Map.Entry<String, Long> hashAndAccession : assignedAccessions.entrySet()) {
-            List<AccessionWrapper<IClusteredVariant, String, Long>> allByAccession;
-            try {
-                allByAccession = clusteredService.getAllByAccession(hashAndAccession.getValue());
-            } catch (AccessionMergedException | AccessionDoesNotExistException | AccessionDeprecatedException cause) {
-                throw new RuntimeException(cause);
-            }
-            if (isMultimap(allByAccession.stream().map(AccessionWrapper::getData).collect(Collectors.toList()))) {
-                multimapHashes.add(hashAndAccession.getKey());
-            }
-        }
-        multimapHashes.forEach(assignedAccessions::remove);
+    private void excludeMultimaps(List<GetOrCreateAccessionWrapper<IClusteredVariant, String, Long>> accessionWrappers) {
+        List<GetOrCreateAccessionWrapper<IClusteredVariant, String, Long>> accessionsNoMultimap =
+                accessionWrappers.stream().filter(x -> !isMultimap(x.getData())).collect(Collectors.toList());
+
+        accessionsNoMultimap.forEach(x -> assignedAccessions.put(x.getHash(), x.getAccession()));
```

Comment:
I'm not sure I expect this statement to be part of the `excludeMultimaps` function, It might just be me though.

Reply:
I was not sure either about that but I agree is better not to include it here. Changed the method, was thinking about just removing this method and doing the filtering directly in `getOrCreateClusteredVariantAccessions` but is probably clearer to keep the method also because of the comment in the header of the `excludeMultimaps` method